packages/edit-post/src/components/header/post-publish-button-or-toggle.js

```diff
@@ -24,54 +24,54 @@ export function PostPublishButtonOrToggle( {
 	isScheduled,
 	togglePublishSidebar,
 } ) {
-	const button = (
-		<PostPublishButton
-			forceIsDirty={ forceIsDirty }
-			forceIsSaving={ forceIsSaving }
-		/>
-	);
-	const toggle = (
-		<PostPublishPanelToggle
-			isOpen={ isPublishSidebarOpened }
-			onToggle={ togglePublishSidebar }
-			forceIsSaving={ forceIsSaving }
-			forceIsDirty={ forceIsDirty }
-		/>
-	);
+	const IS_TOGGLE = 'toggle';
+	const IS_BUTTON = 'button';
+	let component;
 
 	/**
-	 * We want to show a BUTTON when the post status is at the _final stage_
+	 * Conditions to show a BUTTON (publish directly) or a TOGGLE (open publish sidebar):
+	 *
+	 * 1) We want to show a BUTTON when the post status is at the _final stage_
 	 * for a particular role (see https://codex.wordpress.org/Post_Status):
 	 *
 	 * - is published
 	 * - is scheduled to be published
-	 * - is pending and can't be published (but only for viewports >= medium)
+	 * - is pending and can't be published (but only for viewports >= medium).
+	 * 	 Originally, we considered showing a button for pending posts that couldn't be published
+	 * 	 (for example, for an author with the contributor role). Some languages can have
+	 * 	 long translations for "Submit for review", so given the lack of UI real estate available
+	 * 	 we decided to take into account the viewport in that case.
```

Comment:
I find this comment a bit confusing, but I know it was like that when you got here. I'm not really sure what:

> so given the lack of UI real estate available we decided to take into account the viewport in that case

means. 🤷‍♂️

If you can clarify it that'd be great, but no worries if not.

Reply:
Added a link to the PR that originated that change.